src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionAnnotatedTypeFactory.java

```diff
@@ -1,18 +1,164 @@
+    this.bindMethods.add(
+        TreeUtils.getMethod(
+            "com.google.inject.AbstractModule", "bind", processingEnv, "java.lang.Class<T>"));
+
+    this.toMethods.add(
+        TreeUtils.getMethod(
+            "com.google.inject.binder.LinkedBindingBuilder",
+            "to",
+            processingEnv,
+            "java.lang.Class<? extends T>"));
+    this.toMethods.add(
+        TreeUtils.getMethod(
+            "com.google.inject.binder.LinkedBindingBuilder",
+            "to",
+            processingEnv,
+            "com.google.inject.TypeLiteral<? extends T>"));
+    this.toMethods.add(
+        TreeUtils.getMethod(
+            "com.google.inject.binder.LinkedBindingBuilder",
+            "to",
+            processingEnv,
+            "com.google.inject.Key<? extends T>"));
     super.postInit();
   }
 
+  /* Returns true iff the argument is an invocation of AbstractModule.bind.
+   *
+   * @param methodTree the method invocation tree
+   * @return true iff the argument is an invocation of AbstractModule.bind()
+   */
+  private boolean isBindMethod(Tree methodTree) {
+    return TreeUtils.isMethodInvocation(methodTree, this.bindMethods, this.getProcessingEnv());
+  }
+
+  /* Returns true iff the argument is an invocation of AbstractModule.to() */
+  private boolean isToMethod(Tree methodTree) {
+    return TreeUtils.isMethodInvocation(methodTree, this.toMethods, this.getProcessingEnv());
+  }
+
   @Override
   protected void postAnalyze(ControlFlowGraph cfg) {
+
+    System.out.println("--------------------");
+
+    Set<Block> visited = new HashSet<>();
+    Deque<Block> worklist = new ArrayDeque<>();
+    HashMap<Integer, Integer> knownBindings = new HashMap<>();
+
+    Block entry = cfg.getEntryBlock();
+    worklist.add(entry);
+    visited.add(entry);
+
+    while (!worklist.isEmpty()) {
+      Block current = worklist.remove();
+
+      current
+          .getNodes()
+          .forEach(
+              node -> {
+                if (node instanceof MethodInvocationNode) {
+                  MethodInvocationNode methodInvocationNode = (MethodInvocationNode) node;
+
+                  System.out.printf("Method Invocation: %s\n", methodInvocationNode);
+                  System.out.printf("Operands: %s\n\n", methodInvocationNode.getOperands());
+
+                  if (methodInvocationNode.getOperands().size() >= 2) {
+                    MethodAccessNode methodAccessNode = methodInvocationNode.getTarget();
+                    Node fieldAccessNode = methodInvocationNode.getArgument(0);
+
+                    if (isBindMethod(methodInvocationNode.getTree())) {
+                      // Class that is being bound - put in knownBindings
+                      AnnotatedTypeMirror boundClassTypeMirror =
+                          this.getAnnotatedType(fieldAccessNode.getTree());
+
+                      knownBindings.put(boundClassTypeMirror.getUnderlyingTypeHashCode(), null);
```

Comment:
I don't think we want to be storing the underlying type's hashcode here (unless I'm misunderstanding what the specification of `bind()` is). 

Instead, I would expect that here you would get the `@ClassVal` annotation out of `boundClassTypeMirror` (using `getAnnotation()` or one of the related APIs). Then, you could get the argument to that annotation and use that class.

The reasoning here is that the underlying type comes from the regular Java type system, but you want to be reasoning about the values of class literals (which requires the Class Val type system).

Reply:
I'm not able to figure out how to extract the annotation cleanly from an `AnnotationMirror` which is what `getAnnotation()` returns. Instead I opted to use the `getUnderlyingType()` method that exists on the `AnnotatedTypeMirror`. If this isn't good either we can discuss this some more.

Reply:
The problem with using `getUnderlyingType()` is that it produces something different (but that I think coincides in your example). In particular, it returns the Java base type of the underlying object, which means that e.g., if a `.class` object was assigned to a variable with type `Object`, you'd get `Object` back instead of the class.

To get the fields of an `AnnotationMirror`, you usually use one of the methods in `AnnotationUtils`: probably `getElementValueArray`. You can find plenty of examples of how those methods are used in the various CF checkers.